arangosh/Shell/V8ClientConnection.cpp

```diff
@@ -1790,41 +1836,135 @@ v8::Local<v8::Value> V8ClientConnection::requestData(
+  return true;
+}
+
+bool canParseResponse(fu::Response const& response) {
+  return (response.contentType() != fuerte::ContentType::Custom) &&
+    (response.isContentTypeVPack() || response.isContentTypeJSON()) &&
+    (response.contentEncoding() == fuerte::ContentEncoding::Identity) &&
+    (response.payload().size() > 0);
+}
+
+v8::Local<v8::Value> parseReplyBodyToV8(fu::Response const& response,
+                                        v8::Isolate* isolate) {
+  if (response.contentType() == fu::ContentType::VPack) {
+    std::vector<VPackSlice> const& slices = response.slices();
+    return TRI_VPackToV8(isolate, slices[0]);
+  } else if (response.contentType() == fu::ContentType::Json) {
+    auto responseBody = response.payload();
+    try {
+      auto parsedBody = VPackParser::fromJson(
+        reinterpret_cast<char const*>(
+          responseBody.data()),
+        responseBody.size());
+      return TRI_VPackToV8(isolate, parsedBody->slice());
+    } catch (std::exception const& ex) {
+      std::string err("Error parsing the server JSON reply: ");
+      err += ex.what();
+      TRI_CreateErrorObject(isolate, TRI_ERROR_HTTP_CORRUPTED_JSON, err, true);
     }
+  } 
+  return v8::Local<v8::Value>();
+}
+
+v8::Local<v8::Value> translateResultBodyToV8(fu::Response const& response,
+                                             v8::Isolate* isolate) {
+  auto responseBody = response.payload();
+
+  if ((response.contentEncoding() == fuerte::ContentEncoding::Identity) ||
+      response.isContentTypeHtml() ||
+      response.isContentTypeText()) {
+    const char* bodyStr = reinterpret_cast<const char*>(responseBody.data());
+    return TRI_V8_PAIR_STRING(isolate, bodyStr, responseBody.size());
+  } else {
+    V8Buffer* buffer = V8Buffer::New
+      (isolate,
+       static_cast<char const*>(responseBody.data()),
+       responseBody.size());
+    return v8::Local<v8::Object>::New(isolate, buffer->_handle);
   }
-  else if (_forceJson && (req->header.acceptType() == fu::ContentType::VPack)) {
-    req->header.acceptType(fu::ContentType::Json);
+}
+void setResultMessage(v8::Isolate* isolate,
+                      v8::Local<v8::Context> context,
+                      bool isError,
+                      unsigned lastHttpReturnCode,
+                      std::string const& message,
+                      v8::Local<v8::Object> result) {
+    result->Set(context,
+                TRI_V8_STD_STRING(isolate, StaticStrings::Error),
+                v8::Boolean::New(isolate, true)).FromMaybe(isError);
+    result->Set(context,
+                TRI_V8_STD_STRING(isolate, StaticStrings::ErrorNum),
+                v8::Integer::New(isolate, lastHttpReturnCode)).FromMaybe(false);
+    result->Set(context,
+                TRI_V8_STD_STRING(isolate, StaticStrings::ErrorMessage),
+                TRI_V8_STD_STRING(isolate, message)).FromMaybe(false);
+}
+
+void setResultMessage(v8::Isolate* isolate,
```

Comment:
It seems there are 2 `setResultMessage` functions now, one with a message parameter and one without. Can we move them close together in code so this becomes more obvious to the code reader?

Reply:
the one even calls the other?